### ISSUE TITLE
Custom Jest config

### ISSUE THREAD

**Issue by nthtran** [timarney/react-app-rewired#43]
Is it possible to define custom Jest config, not just Webpack?

**Comment by timarney** [comment]
Not yet.

**Comment by timarney** [comment]
Looks like this was merged https://github.com/facebookincubator/create-react-app/pull/1830 with CRA 1.0.0 does that get you what you need?

**Comment by nthtran** [comment]
That only works for a few options. Not all. You'd still need to eject to have full control over Jest config.

**Comment by nicmyers** [comment]
I am also looking for this feature, specifically the ability to overwrite 'transformIgnorePatterns' as it's currently excluding node_modules. transformIgnorePatterns is not one of the four allowable options you can overwrite within CRA 1.0+

**Comment by timarney** [comment]
I won't have any time soon to work on this but it makes sense for this project.

Would be ideal if the jest config could be setup like the others i.e. how the override works require.cache[require.resolve(webpackConfig)].exports.

https://github.com/timarney/react-app-rewired/blob/master/packages/react-app-rewired/scripts/build.js

**Comment by timarney** [comment]
@nicmyers @nthtran does this merged pull request help either of you?

https://github.com/timarney/react-app-rewired/pull/59

**Comment by nthtran** [comment]
Yes I think this would solve it. Will test it out. Thanks @Gregoirevda and @timarney!

**Comment by Gregoirevda** [comment]
You're welcome! I've added a test repo that shows how it is used: https://github.com/Gregoirevda/react-app-rewired-jest-overwrite

Adding jest info in package.json will work now. (config-overrides.js is not used for test, only for dev)

**Comment by nicholasc** [comment]
The last merge broke my basic jest config in my package.json.

Here is the configuration I was using before 1.0.10 and that was working:
```
jest: {
  "collectCoverageFrom": [
    "src/**/*js,
    "!src/index.js",
    "!src/setupTests.js"
  ]
}
```

Since updating, I get the following error, yet my configuration clearly is using an array.

<img width="350" alt="screen shot 2017-07-17 at 12 52 40 pm" src="https://user-images.githubusercontent.com/2378304/28279562-e2e43f7a-6aee-11e7-9b2f-a49e2ad0a2f2.png">


[This](https://github.com/timarney/react-app-rewired/blob/master/packages/react-app-rewired/scripts/utils/createJestConfig.js#L60) is causing the issue. The typeof of a variable will never be an array so arrays in the configuration will always be converted to objects.

I've submitted a PR to fix this [#60](https://github.com/timarney/react-app-rewired/pull/60)

**Comment by timarney** [comment]
Going to close this - can open more specific issues as needed.
